watertap/examples/flowsheets/electrodialysis/electrodialysis_1stack_conc_recirc.py

```diff
@@ -0,0 +1,460 @@
+        "mw_data": {"H2O": 18e-3, "Na_+": 23e-3, "Cl_-": 35.5e-3},
+        "elec_mobility_data": {("Liq", "Na_+"): 5.19e-8, ("Liq", "Cl_-"): 7.92e-8},
+        "diffusivity_data": {("Liq", "Na_+"): 1.33e-9, ("Liq", "Cl_-"): 2.03e-9},
+        "charge": {"Na_+": 1, "Cl_-": -1},
+    }
+    m.fs.properties = MCASParameterBlock(**ion_dict)
+    m.fs.costing = WaterTAPCosting()
+    m.fs.feed = Feed(property_package=m.fs.properties)
+    m.fs.sepa0 = Separator(
+        property_package=m.fs.properties,
+        outlet_list=["to_dil_in", "to_conc_in0"],
+    )
+    m.fs.mix0 = Mixer(
+        property_package=m.fs.properties,
+        energy_mixing_type=MixingType.none,
+        momentum_mixing_type=MomentumMixingType.none,
+        inlet_list=["from_feed", "from_conc_out"],
+    )
+
+    m.fs.pump0 = Pump(property_package=m.fs.properties)
+    m.fs.pump0.ratioP_calculation.deactivate()
+    m.fs.pump1 = Pump(property_package=m.fs.properties)
+    m.fs.pump0.ratioP_calculation.deactivate()
+
+    # Add electrodialysis (ED) stacks
+    m.fs.EDstack = Electrodialysis1D(
+        property_package=m.fs.properties,
+        operation_mode=ElectricalOperationMode.Constant_Voltage,
+        finite_elements=20,
+        has_pressure_change=True,
+        has_nonohmic_potential_membrane=True,
+        has_Nernst_diffusion_layer=True,
+        limiting_current_density_method=LimitingCurrentDensityMethod.Theoretical,
+        pressure_drop_method=PressureDropMethod.Darcy_Weisbach,
+        hydraulic_diameter_method=HydraulicDiameterMethod.spacer_specific_area_known,
+        friction_factor_method=FrictionFactorMethod.Gurreri,
+    )
+    m.fs.sepa1 = Separator(
+        property_package=m.fs.properties,
+        outlet_list=["to_disp", "to_conc_in1"],
+    )
+    m.fs.prod = Product(property_package=m.fs.properties)
+    m.fs.disp = Product(property_package=m.fs.properties)
+
+    # touch necessary variables to ensure they are constructed and solved
+    m.fs.prod.properties[0].flow_vol_phase[...]
+    m.fs.disp.properties[0].flow_vol_phase[...]
+
+    # costing
+    m.fs.EDstack.costing = UnitModelCostingBlock(flowsheet_costing_block=m.fs.costing)
+    m.fs.pump0.costing = UnitModelCostingBlock(flowsheet_costing_block=m.fs.costing)
+    m.fs.pump1.costing = UnitModelCostingBlock(flowsheet_costing_block=m.fs.costing)
+    m.fs.costing.cost_process()
+    m.fs.costing.add_annual_water_production(
+        m.fs.prod.properties[0].flow_vol_phase["Liq"]
+    )
+    m.fs.costing.add_LCOW(m.fs.prod.properties[0].flow_vol)
+    m.fs.costing.add_specific_energy_consumption(
+        m.fs.prod.properties[0].flow_vol_phase["Liq"]
+    )
+
+    # add extra variables and constraints
+    m.fs.recovery_vol_H2O = Var(
+        initialize=0.5,
+        bounds=(0, 1),
+        domain=NonNegativeReals,
+        units=pyunits.dimensionless,
+        doc="flowsheet level water recovery calculated by volumeric flow rate",
+    )
+    m.fs.mem_area = Var(
```

Comment:
Any reason why this created at the flowsheet level rather than the unit model?

Reply:
The unit does not need this var for anything but the flowsheet need it for reporting data.